Checkpoint/checkpoint/ingest/parser.yml

```
@@ -0,0 +1,294 @@
+          action.outcome_reason: "{{checkpoint.message.msg}}"
+          action.properties.__policy_id_tag: "{{checkpoint.message.get('__policy_id_tag')}}"
+          action.properties.community: "{{checkpoint.message.community}}"
+          action.properties.destination_key_id: "{{checkpoint.message.dstkeyid}}"
+          action.properties.encryption_methods: "{{checkpoint.message.methods}}"
+          action.properties.encryption_scheme: "{{checkpoint.message.scheme}}"
+          action.properties.flags: "{{checkpoint.message.flags}}"
+          action.properties.icmp_code: "{{checkpoint.message.icmp_code}}"
+          action.properties.icmp_message: "{{checkpoint.message.icmp}}"
+          action.properties.icmp_type: "{{checkpoint.message.icmp_type}}"
+          action.properties.ike_ids: "{{checkpoint.message.ike_ids}}"
+          action.properties.ike_initiator_cookie: "{{checkpoint.message.cookiei}}"
+          action.properties.ike_mode: "{{checkpoint.message.ike}}"
+          action.properties.ike_responder_cookie: "{{checkpoint.message.cookier}}"
+          action.properties.loguid: "{{checkpoint.message.loguid}}"
+          action.properties.nat_addtnl_rulenum: "{{checkpoint.message.nat_addtnl_rulenum}}"
+          action.properties.nat_rulenum: "{{checkpoint.message.nat_rulenum}}"
+          action.properties.observer_type: "{{checkpoint.message.DeviceProduct}}"
+          action.properties.origin: "{{checkpoint.message.origin}}"
+          action.properties.origin_sic_name: "{{checkpoint.message.origin_sic_name}}"
+          action.properties.originsicname: "{{checkpoint.message.originsicname.replace('\\\\', '')}}"
+          action.properties.product: "{{checkpoint.message.product}}"
+          action.properties.reject_category: "{{checkpoint.message.reject_category}}"
+          action.properties.rule_name: "{{checkpoint.message.get('Rule Name')}}"
+          action.properties.source_key_id: "{{checkpoint.message.srckeyid}}"
+          action.properties.subproduct: "{{checkpoint.message.fw_subproduct}}"
+          action.properties.vpn_feature_name: "{{checkpoint.message.vpn_feature_name}}"
+          action.properties.vpn_peer_gateway: "{{checkpoint.message.peer_gateway}}"
+          destination.address: "{{checkpoint.message.dst}}"
+          destination.domain: "{{checkpoint.message.dhost}}"
+          destination.ip: "{{checkpoint.message.dst}}"
+          destination.nat.ip: "{{checkpoint.message.destinationTranslatedAddress or checkpoint.message.xlatedst}}"
+          destination.nat.port: "{{checkpoint.message.destinationTranslatedPort or checkpoint.message.xlatedport}}"
+          destination.port: "{{checkpoint.message.service or checkpoint.message.dpt}}"
+          destination.user.name: "{{checkpoint.message.duser}}"
+          event.code: "{{checkpoint.message.DeviceVendor or checkpoint.message.loguid or checkpoint.message.DeviceEventClassID}}"
```

I suggest using `DeviceEventClassId` first and remove `DeviceVendor`
```suggestion
          event.code: "{{checkpoint.message.DeviceEventClassID or checkpoint.message.loguid}}"
```